src/main/java/au/com/codeka/carrot/Scope.java

```diff
@@ -36,12 +38,17 @@ public void pop() {
     stack.pop();
   }
 
-  /** @return The current bindings off the top of the stack. */
-  public Map<String, Object> peek() {
-    return stack.peek();
-  }
 
   /**
+   * Extend the current bindings with the given one.
+   *
+   * @param bindings The new {@link Bindings}.
+   */
+  public void extendCurrent(Bindings bindings)
+  {
```

Comment:
This one on the previous line, too. I won't bother commenting for the other places, but you get the idea :)